videostore/utils.go

```diff
@@ -228,42 +258,81 @@ func formatDateTimeToString(dateTime time.Time) string {
 }
 
 // matchStorageToRange returns a list of files that fall within the provided time range.
+// The matcher assumes that the input files list is sorted by start time.
 // Includes trimming video files to the time range if they overlap.
 func matchStorageToRange(files []string, start, end time.Time, logger logging.Logger) []string {
 	var matchedFiles []string
+	// Cache of the first matched video file's width, height, and codec
+	// to ensure every video in the matched files set have the same params.
+	firstSeenVideoInfo := videoInfo{
+		width:  0,
+		height: 0,
+		codec:  "",
+	}
 	for _, file := range files {
-		dateTime, err := extractDateTimeFromFilename(file)
+		fileStartTime, err := extractDateTimeFromFilename(file)
 		if err != nil {
 			logger.Debugf("failed to extract datetime from filename: %s, error: %v", file, err)
 			continue
 		}
-		duration, err := getVideoDuration(file)
-		if err != nil {
-			logger.Debugf("failed to get video duration for file: %s, error: %v", file, err)
-			continue
+		if fileStartTime.After(end) {
+			logger.Debugf("Skipping file %s and winding down matcher. File starts after end time (start=%v, end=%v)", file, start, end)
+			break
 		}
-		fileEndTime := dateTime.Add(duration)
-		// Check if the file's time range intersects with [start, end)
-		if dateTime.Before(end) && fileEndTime.After(start) {
+		estimatedFileEndTime := fileStartTime.Add(time.Duration(maxSegmentLength+keyFrameIntervalBuffer) * time.Second)
+		// Check if the segment file's time range intersects
+		// with the match request time range [start, end)
+		if fileStartTime.Before(end) && estimatedFileEndTime.After(start) {
+			videoFileInfo, err := getVideoInfo(file)
+			if err != nil {
+				logger.Debugf("failed to get video duration for file: %s, error: %v", file, err)
+				continue
+			}
+			actualFileEndTime := fileStartTime.Add(videoFileInfo.duration)
+			// If the real file end time is exclusively before the start time, skip the file
+			if !actualFileEndTime.After(start) {
+				logger.Debugf(
+					"Skipping file %s. File ends before start time (end=%v, start=%v)",
+					file, actualFileEndTime, start,
+				)
+				continue
+			}
+			// If the first video file in the matched set, cache the width, height, and codec
+			cacheFirstVid(&firstSeenVideoInfo, videoFileInfo)
```

Comment:
this isn't very idomatic go

try this instead:
```golang
var firstSeenVideoInfo videoInfo
firstSeenVideoInfoInitialized
...

if !firstSeenVideoInfoInitialized {
    firstSeenVideoInfo = videoFileInfo
}
```

Reply:
Moved to initialize with `var firstSeenVideoInfo videoInfo
firstSeenVideoInfoInitialized`

But, do not think i can simply check if struct is filled out with boolean check `if !firstSeenVideoInfoInitialized`